crates/ironrdp-rdpdr/src/pdu/efs.rs

```diff
@@ -264,3 +163,479 @@ impl From<ClientNameRequestUnicodeFlag> for CharacterSet {
         }
     }
 }
+
+/// [2.2.2.7 Server Core Capability Request (DR_CORE_CAPABILITY_REQ)](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpefs/702789c3-b924-4bc2-9280-3221bc7d6797)
+/// [2.2.2.8 Client Core Capability Response (DR_CORE_CAPABILITY_RSP)](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpefs/f513bf87-cca0-488a-ac5c-18cf18f4a7e1)
+#[derive(Debug)]
+pub struct CoreCapability {
+    pub num_capabilities: u16,
```

Comment:
It’s better to not keep this field because:
- it is redundant with `capabilities.len()`
- it’s easy to end up with `num_capabilities != capabilities.len()`

When encoding, you can use the `cast_length!` macro to handle the usize → u16 conversion properly:
```rust
dst.write_u16(cast_length!("CoreCapability", "numCapabilities", self.capabilities.len()?);
```
(first parameter is not required when `<Self as Pdu>::NAME` is defined)

In the unlikely event of `self.capabilities.len()` value being too big to fit a `u16`, the final error will look like:
```
[CoreCapability] invalid `numCapabilities`: too many elements, caused by: out of range integral type conversion attempted
```

Reply:
554e22a7ed8bc3c665e9ef0045c77ff9021b1450